src/prefect/results.py

```diff
@@ -255,8 +264,15 @@ async def _read(self, key: str) -> "ResultRecord":
         if self.result_storage is None:
             self.result_storage = await get_default_result_storage()
 
-        content = await self.result_storage.read_path(f"{key}")
-        return ResultRecord.deserialize(content)
+        if self.metadata_storage is not None:
+            metadata_content = await self.metadata_storage.read_path(f"{key}")
+            result_content = await self.result_storage.read_path(f"{key}")
```

Comment:
two questions:
- `f"{key}"` looks odd, why not `key`?
- I think that the result_storage key should be read from the metadata content, and not assumed to be living at the same key in the situation where they are stored separately; that separation I think is important for allowing Prefect to reason about results for which we don't own the key logic

Reply:
Makes sense! I updated the result storage key logic in https://github.com/PrefectHQ/prefect/pull/15214/commits/7ff0a184cc828da249c08b49d1e1ab9bc767329c